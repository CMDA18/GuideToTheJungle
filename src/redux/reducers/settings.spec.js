import Settings from './settings'

const probeAction = {type: '@@PROBE'}

describe('Settings Reducer', () => {
  it('returns initial state when no state is provided', () => {
    expect(Settings(undefined, probeAction)).toMatchSnapshot()
  })

  it('handles the TEST action', () => {
    expect(Settings(undefined, {
      type: 'TEST',
      value: 'testValue'
    })).toMatchSnapshot()
  })

  it('handles the TEST_SUCCESS action', () => {
    expect(Settings(undefined, {
      type: 'TEST_SUCCESS',
      value: 'testValue'
    })).toMatchSnapshot()
  })

  it('handles the PRELOAD action', () => {
    expect(Settings(undefined, {
      type: 'PRELOAD',
      value: 'testValue'
    })).toMatchSnapshot()
  })

  it('handles the PRELOAD_SUCCESS action', () => {
    expect(Settings(undefined, {
      type: 'PRELOAD_SUCCESS',
      value: 'testValue'
    })).toMatchSnapshot()
  })
})
