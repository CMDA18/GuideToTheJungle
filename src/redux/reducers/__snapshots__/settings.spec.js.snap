// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Reducer handles the PRELOAD action 1`] = `
Object {
  "preload": "pre:testValue",
  "test": "hoi",
}
`;

exports[`Settings Reducer handles the PRELOAD_SUCCESS action 1`] = `
Object {
  "preload": "pre:done:testValue",
  "test": "hoi",
}
`;

exports[`Settings Reducer handles the TEST action 1`] = `
Object {
  "newTest": "sync:testValue",
  "test": "hoi",
}
`;

exports[`Settings Reducer handles the TEST_SUCCESS action 1`] = `
Object {
  "newTest": "testValue",
  "test": "hoi",
}
`;

exports[`Settings Reducer returns initial state when no state is provided 1`] = `
Object {
  "test": "hoi",
}
`;
